src/server/endpoints/track-video-history-endpoint.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const usersDb = require('../collections/users-db');
+const tokensDb = require('../collections/tokens-db');
+
+function updateErrorhandle(res, result) {
+  if (result === 'Upadate Failed') {
+    res.status(500).json({'error': 'Update Failed'});
+    return;
+  }
+  if (result === undefined) {
+    res.status(500).json({'error': 'Unable connect database'});
+    return;
+  }
+}
+
+function trackVideoHistory(req, res) {
+  let token = req.get('Authorization');
+  let userId;
+
+  if (token === undefined) {
+    res.status(400).json({'error': 'unauthorization'});
+    return;
+  }
+  tokensDb.getToken(token, (tokenInfos) => {
+    userId = tokenInfos.userId;
+    let videoId = req.params.videoId;
+
+    usersDb.findUserInfoById(userId, (userInfos) => {
+      if (userInfos.email === undefined) {
+        res.status(404).json({'error': 'not found'});
+        return;
+      }
+      let existuser = userInfos.history.filter(function(value, index, array) {
```

Comment:
when we are talking about history we should track each view of a given video. Maybe the only thing what you can check if the video was the last view or not. So if we have 'a', 'b', and 'c' video then
[a, b, a, c, a, b] is a valid history but
[a, a, b, c, a, b, b] is not because [a, a] and [b, b] should be on entry.